enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -6,6 +6,57 @@ use sgx_types::{ssa::StateSaveArea, tcs::Tcs};
 
 use core::ops::Range;
 
+// ===== Temporary nolibc functions ================
+
+enumerate::enumerate! {
+    #[derive(Copy, Clone)]
+    pub enum ArchPrctlTask: u64 {
+        ArchSetGs = 0x1001,
+        ArchSetFs = 0x1002,
+        ArchGetFs = 0x1003,
+        ArchGetGs = 0x1004,
+    }
+}
+
+/// # Safety
+///
+/// All libc functions are unsafe.
+#[cfg_attr(not(test), no_mangle)]
+pub unsafe extern "C" fn _memcpy(dst: *mut u8, src: *const u8, n: usize) -> *mut u8 {
+    for i in 0..n {
+        *dst.add(i) = *src.add(i);
+    }
+
+    dst
+}
+/// # Safety
+///
+/// All libc functions are unsafe.
+#[cfg_attr(not(test), no_mangle)]
+pub unsafe extern "C" fn _memset(dst: *mut u8, src: u8, n: usize) -> *mut u8 {
+    for i in 0..n {
+        *dst.add(i) = src;
+    }
+
+    dst
+}
+
+/// # Safety
+///
+/// All libc functions are unsafe.
+#[cfg_attr(not(test), no_mangle)]
+pub unsafe extern "C" fn _bcmp(l: *const u8, r: *const u8, n: usize) -> usize {
+    let mut neq = 0;
+
+    for i in 0..n {
+        neq |= (*l.add(i) != *r.add(i)) as usize;
+    }
+
+    neq
+}
```

Comment:
These should go into their own module. A top-level `libc` module is probably best. Also, remove the word temporary (don't forget the commit message). They might not be temporary.

Reply:
Just FYI, what I tried to do initially is put them in their own module - a separate file in `enarx-keep-sgx-shim` and then `pub mod` for that in `main.rs`, but it gave me the same linking error until I put them all in this file.

Reply:
@lkatalin Probably because you didn't remove them from `nolibc`.

Reply:
Hmm, okay. I will try once more.